apps/etterlatte-behandling/src/main/kotlin/institusjonsopphold/InstitusjonsoppholdDao.kt

```diff
@@ -1,23 +1,54 @@
 package no.nav.etterlatte.institusjonsopphold
 
+import com.fasterxml.jackson.module.kotlin.readValue
+import no.nav.etterlatte.libs.common.behandling.JaNei
+import no.nav.etterlatte.libs.common.grunnlag.Grunnlagsopplysning
+import no.nav.etterlatte.libs.common.objectMapper
+import no.nav.etterlatte.libs.database.singleOrNull
 import java.sql.Connection
 import java.util.*
 
 class InstitusjonsoppholdDao(private val connection: () -> Connection) {
-    fun lagreInstitusjonsopphold(sakId: Long, institusjonoppholdBegrunnelse: InstitusjonsoppholdBegrunnelse) {
+    fun lagreInstitusjonsopphold(
+        sakId: Long,
+        saksbehandler: Grunnlagsopplysning.Saksbehandler,
+        institusjonoppholdBegrunnelse: InstitusjonsoppholdBegrunnelse
+    ) {
         with(connection()) {
             val statement = prepareStatement(
                 "INSERT INTO institusjonsoppholdhendelse(id, sak_id, kanGiReduksjon, kanGiReduksjonTekst," +
-                    "merEnnTreMaaneder, merEnnTreMaanederTekst) " +
-                    "VALUES(?::UUID, ?, ?, ?, ?, ?)"
+                    "merEnnTreMaaneder, merEnnTreMaanederTekst, saksbehandler, grunnlagsendringshendelse_id) " +
+                    "VALUES(?::UUID, ?, ?, ?, ?, ?, ?, ?::UUID)"
             )
             statement.setString(1, UUID.randomUUID().toString())
             statement.setLong(2, sakId)
             statement.setString(3, institusjonoppholdBegrunnelse.kanGiReduksjonAvYtelse.toString())
             statement.setString(4, institusjonoppholdBegrunnelse.kanGiReduksjonAvYtelseBegrunnelse)
             statement.setString(5, institusjonoppholdBegrunnelse.forventetVarighetMerEnn3Maaneder.toString())
             statement.setString(6, institusjonoppholdBegrunnelse.forventetVarighetMerEnn3MaanederBegrunnelse)
+            statement.setString(7, saksbehandler.toJson())
+            statement.setString(8, institusjonoppholdBegrunnelse.grunnlagsEndringshendelseId)
             statement.executeUpdate()
         }
     }
+
+    fun hentBegrunnelse(grunnlagsEndringshendelseId: String): InstitusjonsoppholdBegrunnelseMedSaksbehandler? {
+        return with(connection()) {
+            val statement = prepareStatement(
+                "SELECT * from institusjonsoppholdhendelse" +
+                    " WHERE grunnlagsendringshendelse_id = ?::UUID"
+            )
+            statement.setString(1, grunnlagsEndringshendelseId)
+            statement.executeQuery().singleOrNull {
+                InstitusjonsoppholdBegrunnelseMedSaksbehandler(
+                    kanGiReduksjonAvYtelse = JaNei.valueOf(getString("kanGiReduksjon")),
```

Comment:
Lurer på om vi bør ha ein type her for kombinasjonen av JaNei og begrunnelse, dei heng jo saman, og det kan vi gjerne modellere inn eksplisitt

Reply:
Høres ut fornuftig ut. Det er kanskje ingen steder vi kun bruker `JaNei`.

Reply:
Tas i https://jira.adeo.no/browse/EY-2335